btc.com/src/endpoint/balance.ts

```diff
@@ -0,0 +1,94 @@
+import objectPath from 'object-path'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { AdapterRequest } from '@chainlink/types'
+import { Config, DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { CoinType, ChainType } from '.'
+
+export const Name = 'balance'
+
+type Address = {
+  address: string
+  coin?: CoinType
+  chain?: ChainType
+  balance?: number
+}
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+type ResponseWithResult = {
+  response?: any
+  result?: Address
+}
+
+const WARNING_NO_OPERATION = 'No Operation: only btc mainnet is supported by BTC.com adapter'
+const WARNING_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+const getBalanceURI = (address: string) => `/v3/address/${address}`
+
+const toBalances = async (config: Config, addresses: Address[]): Promise<ResponseWithResult[]> =>
+  Promise.all(
+    addresses.map(async (addr: Address) => {
+      if (!addr.address) return { ...addr, warning: WARNING_NO_OPERATION_MISSING_ADDRESS }
```

Comment:
I think we should change this to an error as `address` should be required. I see this is also an issue in `blockchain.com` & `blockcypher` adapters. Please also throw an error there.

There is a difference between adapter does not support or can't process something (warning or error depending on `strict` flag) vs. adapter could not process it even if it supported that capability (error).

Reply:
I'm thinking of adding a method to Requester called `warnOrThrow` that will handle the strict flag. 

Any ideas on a better implementation?